lib/pbench/cli/agent/commands/results/__init__.py

```diff
@@ -1,94 +1,206 @@
+"pbench-move-results"
+
 import os
-import requests
+import click
+
+# import requests
 import shutil
+
 import sys
 import tempfile
 
 from pbench.agent import PbenchAgentConfig
-from pbench.agent.results import MakeResultTb, CopyResultTb
+from pbench.agent.base import BaseCommand
+from pbench.agent.results import (
+    MakeResultTb,
+    CopyResultTb,
+    FileUploadError,
+)
+from pbench.common.exceptions import BadMDLogFormat
 from pbench.common.logger import get_pbench_logger
+from pbench.cli.agent import pass_cli_context
+from pbench.cli.agent.options import common_options
 
+PROG = "pbench-move-results"
 
-def move_results(ctx, _user, _prefix, _show_server):
-    config = PbenchAgentConfig(ctx["args"]["config"])
-    logger = get_pbench_logger("pbench", config)
 
-    controller = os.environ.get("_pbench_full_hostname")
-    if not controller:
-        logger.error("Missing controller name (should be 'hostname -f' value)")
-        sys.exit(1)
+class MoveResults(BaseCommand):
+    def __init__(self, context):
+        super().__init__(context)
 
-    results_webserver = config.results.get("webserver")
-    if not results_webserver:
-        logger.error(
-            "No web server host configured from which we can fetch the FQDN of the host to which we copy/move results"
-        )
-        logger.debug("'webserver' variable in 'results' section not set")
+    def execute(self):
+        config = PbenchAgentConfig(self.context.config)
+        logger = get_pbench_logger("pbench-agent", config)
 
-    server_rest_url = config.results.get("server_rest_url")
-    response = requests.get(f"{server_rest_url}/host_info")
-    if response.status_code not in [200, 201]:
-        logger.error(
-            "Unable to determine results host info from %s/host_info", server_rest_url
-        )
-        sys.exit(1)
-    if response.text.startswith("MESSAGE"):
-        message = response.text.split("===")[1]
-        logger.info("*** Message from sysadmins of %s:", results_webserver)
-        logger.info("***\n*** %s", message)
-        logger.info("***\n*** No local actions taken.")
-        sys.exit(1)
-    results_path_prefix = response.text.split(":")[1]
-    if not results_path_prefix:
-        logger.error(
-            "fetch results host info did not contain a path prefix: %s", response.text
-        )
-        sys.exit(1)
+        if self.context.user:
+            print(f"{PROG}: Option --user is no longer being used", file=sys.stderr)
+        if self.context.prefix:
+            print(f"{PROG}: Option --prefix is no longer being used", file=sys.stderr)
+
+        results_webserver = config.get("results", "webserver")
+        if not results_webserver:
+            logger.error(
+                "No web server host configured from which we can fetch the FQDN of the host to which we copy/move results"
+            )
+
+        # FIXME: Pytest Failure
```

Comment:
You mentioned earlier in response to Webb's comment that you were having Pytest problems with this; can you describe them?

It's also a bit silly that we still need to do a `GET` just to find out the server status before doing the `PUT`. I really think that we need to factor this status check into the `PUT` and it would simply return this as the failure `message` if the server isn't accepting datasets. But that's another PR...

Reply:
it is returning as status 1 due to sys.exit(1) in line 49.